d2/src/main/java/com/linkedin/d2/balancer/clients/TrackerClientFactory.java

```diff
@@ -66,6 +64,17 @@ public static TrackerClient createTrackerClient(URI uri,
     return createTrackerClient(uri, uriProperties, serviceProperties, loadBalancerStrategyName, transportClient, SystemClock.instance());
   }
 
+  public static TrackerClient createTrackerClient(URI uri,
+      UriProperties uriProperties,
+      ServiceProperties serviceProperties,
+      String loadBalancerStrategyName,
+      TransportClient transportClient,
+      boolean loadBalanceStreamException)
+  {
+    return createTrackerClient(uri, uriProperties, serviceProperties, loadBalancerStrategyName, transportClient,
+        SystemClock.instance(), loadBalanceStreamException);
```

Comment:
Can we just propagate `D2ClientConfig` everywhere to avoid having to do this each time a change needs to be made to tracker client, state updater, degrader, etc.

Reply:
It will be weird for inner components like DegraderImpl.Config, DegraderTrackerClientImpl, etc. to hold a D2ClientConfig, which is the upper/larger layer that's supposed to control everything in d2 client.

Reply:
In that case, can a new config object be created for the LB components such as tracker clients, state updater, etc.? It is essential for new code to be open for extensions.

Reply:
adding a StateUpdater.LoadBalanceConfig makes sense. But adding one to TrackerClientFactory will be hard since the createTrackerClient method is static.

Reply:
Added TODOs for optimizing config management later. 
Since we don't have bandwidth to address this now, and this PR is for an Incident action item, it doesn't worth holding off the merge for this optimization now.